src/javascript/app_2/App/Components/Form/input_field.jsx

```diff
@@ -30,9 +30,9 @@ const InputField = ({
     const changeValue = (e) => {
         if (type === 'number') {
             const is_empty = !e.target.value || e.target.value === '';
-            const signed_regex = is_signed ? '[\\+-]?' : '';
+            const signed_regex = is_signed ? '(?!^\\-0$)(?!^\\-$)(?!^\\+$)^[\\+-]?' : '^';
 
-            const is_number = new RegExp(`^${signed_regex}(\\d*)?${is_float ? '(\\.\\d+)?' : ''}(?<=\\d)(?<!-0)$`)
+            const is_number = new RegExp(`${signed_regex}(\\d*)?${is_float ? '(\\.\\d+)?' : ''}$`)
```

Comment:
@easteregg 
it asserts empty string `''` as a valid number.

Reply:
@easteregg 
It still returns true when the `e.target.value` is an empty string.

Reply:
Can't reproduce this.

Reply:
@easteregg 
> Can't reproduce this.

when `is_signed` is false the value of `signed_regex` will be `^` and the regex in line 35 will be `/^(\d*)?$/` or `/^(\d*)?(\.\d+)?$/`.

Reply:
This line is handling the empty string, `const is_empty = !e.target.value || e.target.value === '';` 
We don't need to repeat ourselves.

Reply:
I can remove the second rule that lookahead and denies empty string in signed values as well. didn't notice this line exists before.